doc/encoding.rdoc

```diff
@@ -132,7 +132,232 @@ returns the \Encoding of the concatenated string, or +nil+ if incompatible:
+
+The encoding for a string may be changed:
+
+  s = "R\xC3\xA9sum\xC3\xA9"     # => "Résumé"
+  s.encoding                     # => #<Encoding:UTF-8>
+  s.force_encoding('ISO-8859-1') # => "R\xC3\xA9sum\xC3\xA9"
+  s.encoding                     # => #<Encoding:ISO-8859-1>
+
+Changing the assigned encoding does not alter the content of the string;
+it changes only the way the content is to be interpreted:
+
+  s                         # => "R\xC3\xA9sum\xC3\xA9"
+  s.force_encoding('UTF-8') # => "Résumé"
+
+The actual content of a string may also be altered;
+see {Transcoding a String}[#label-Transcoding+a+String].
+
+Here are a couple of useful query methods:
+
+  s = "abc".force_encoding("UTF-8")         # => "abc"
+  s.ascii_only?                             # => true
+  s = "abc\u{6666}".force_encoding("UTF-8") # => "abc晦"
+  s.ascii_only?                             # => false
+
+  s = "\xc2\xa1".force_encoding("UTF-8") # => "¡"
+  s.valid_encoding?                      # => true
+  s = "\xc2".force_encoding("UTF-8")     # => "\xC2"
+  s.valid_encoding?                      # => false
+
+=== \Symbol and \Regexp Encodings
+
+The string stored in a Symbol or Regexp object also has an encoding.
+which is returned, respectively, by methods Symbol#encoding or Regexp#encoding;
+the default encoding for these, however, is US-ASCII, not UTF-8:
+
+  sym = :foo   # => :foo
+  sym.encoding # => #<Encoding:US-ASCII>
+  re = /foo/   # => /foo/
+  re.encoding  # => #<Encoding:US-ASCII>
+
+=== Filesystem \Encoding
+
+The filesystem encoding is the default \Encoding for a string from the filesystem:
+
+    Encoding.find("filesystem") # => #<Encoding:UTF-8>
+
+=== Locale \Encoding
+
+The locale encoding is the default encoding for a string from the environment,
+other than from the filesystem:
+
+  Encoding.find('locale') # => #<Encoding:IBM437>
+
+=== \IO Encodings
+
+An IO object (an input/output stream), and by inheritance a File object,
+has at least one, and sometimes two, encodings:
+
+- Its _external_ _encoding_ identifies the encoding of the stream.
+- Its _internal_ _encoding_, if not +nil+, specifies the encoding
+  to be used for the string constructed from the stream.
+
+==== External \Encoding
+
+Bytes read from the stream are decoded into characters via the external encoding;
+by default (that is, if the internal encoding is +nil),
+those characters become a string whose encoding is set to the external encoding.
+
+The default external encoding is:
+
+- UTF-8 for a text stream:
+
+    # File.
+    f = File.open('t.txt', 'rt')
+    f.external_encoding  # => #<Encoding:UTF-8>
+
+    # IO.
+    fd = IO.sysopen('t.txt', 'rt')
+    io = IO.new(fd)
+    io.external_encoding # => #<Encoding:UTF-8>
+
+- ASCII-8BIT for a binary stream:
+
+    f = File.open('t.rus', 'rb')
+    f.external_encoding  # => #<Encoding:ASCII-8BIT>
+
+    # IO.
+    fd = IO.sysopen('t.rus', 'rb')
+    io = IO.new(fd)
+    io.external_encoding # => #<Encoding:UTF-8>
```

Comment:
This example is interesting, as it shows that IO.new doesn't inherit encoding. If you know how file descriptors work, this is not surprising, but it's probably worth mentioning the `IO.new` behavior separately.

Reply:
I have removed  the example, in keeping with changes elsewhere.  Doesn't this mention belong over at IO.new?

Reply:
I think it makes sense to mention this in the IO.new documentation.